File: 3/3-ButtonLabel/3-ButtonLabel/ViewController.m
//
//  ViewController.m
//  3_ButtonLabel
//
//  Created by dejin on 2017/5/19.
//  Copyright © 2017年 dejin. All rights reserved.
//

#import "ViewController.h"

@interface ViewController ()

@end

@implementation ViewController

- (IBAction)Click:(id)sender {
    NSString *title = [sender titleForState:UIControlStateNormal];
    NSString *plainText = [NSString stringWithFormat:@"%@ button pressed.", title];
    //_statusLabel.text = plainText;
    NSMutableAttributedString *styledText = [[NSMutableAttributedString alloc] initWithString:plainText];
    NSDictionary *attributes =
  @{
    NSFontAttributeName : [UIFont boldSystemFontOfSize:_statusLabel.font.pointSize]
    };
    NSRange nameRange = [plainText rangeOfString:title];
    [styledText setAttributes:attributes range:nameRange];
    _statusLabel.attributedText = styledText;
    
    
}

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view, typically from a nib.
}


- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}


@end

File: 3/3-ButtonLabelSwift/3-ButtonLabelSwift/ViewController.swift
//
//  ViewController.swift
//  3_ButtonLabelSwift
//
//  Created by dejin on 2017/5/19.
//  Copyright © 2017年 dejin. All rights reserved.
//

import UIKit

class ViewController: UIViewController {

    @IBOutlet weak var statusLabel: UILabel!
    
    @IBAction func Click(_ sender: UIButton) {
        let title = sender.title(for: .normal)!
        let plainText = "\(title) button pressed."
        //statusLabel.text = plainText
        let styledText = NSMutableAttributedString(string: plainText)
        let attributes = [NSFontAttributeName:UIFont.boldSystemFont(ofSize: statusLabel.font.pointSize)]
        let nameRange = (plainText as NSString).range(of: title)
        styledText.setAttributes(attributes, range: nameRange)
        statusLabel.attributedText = styledText
        
    }

    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view, typically from a nib.
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }


}

